core/walker.go

```diff
@@ -3,11 +3,11 @@ package core
 import "context"
 
 type SourceVisitor interface {
-	VisitFile(Language, File) error
+	VisitFile([]Language, File) error
```

Comment:
This seems to introduce the ambiguity - `File` is for which `Language` since there are multiple languages?